packages/gatsby/src/utils/merge-gatsby-config.ts

```diff
@@ -33,7 +19,7 @@ module.exports = (a, b) => {
  *  - { resolve: `gatsby-plugin-name`, options: {} }
  * are all considered equal
  */
-const normalizePluginEntry = entry =>
+const normalizePluginEntry = (entry: PluginEntry): PluginEntry =>
```

Comment:
While this is true and likely won't trip up type checking - return of this function is fully formed so it can't return string or object without `options` (related to my other comment)
```
  {
      resolve: string
      options: Record<string, unknown>
    }
```

It would be good to tighten return type